src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/UseLiteralsWhereAppropriate.cs

```diff
@@ -72,6 +72,7 @@ public override void Initialize(AnalysisContext context)
                         lastField == null ||
                         lastField.IsConst ||
                         !lastField.IsReadOnly ||
+                        (lastField.Type.Name == lastField.Name && fieldInitializerValue.DescendantsAndSelf().Any(d => d is IFieldReferenceOperation field && field.Field.Type.Equals(lastField.Type, SymbolEqualityComparer.Default))) ||
```

Comment:
I always forget if/when the `Type` can be null for field symbols and field reference operations. A more defensive check would be:
```suggestion
                        (lastField.Type?.Name == lastField.Name && fieldInitializerValue.DescendantsAndSelf().Any(d => d is IFieldReferenceOperation field && lastField.Type.Equals(field.Field.Type, SymbolEqualityComparer.Default))) ||
```

Reply:
Good idea. Done.